### PR TITLE
Lock hosts

### PR DESCRIPTION
**Pull request by cornelius**
This pull request implements locking of hosts for integration tests. It provides a command `pennyworth host lock <host_name>`, which acquires a lock for the host. The spec helpers do that implicitly when using `start_system` with the `:host` option.

To set up pennyworth for testing with existing hosts run `pennyworth setup http://machintegration` once. This gets the configuration data from our internal CI host.

An example for an integration test making use of the new feature is:

```ruby
require_relative "integration_spec_helper"

describe "start_system with :host option" do
  it "starts system where commands can be executed" do
    system = start_system(host: "my_test")
    out = system.run_command("cat", "/etc/SuSE-release", stdout: :capture)
    expect(out).to eq("SUSE Linux Enterprise Server 11 (x86_64)\nVERSION = 11\nPATCHLEVEL = 3\n")
  end
end
```

### CODE REVIEW

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
( Success: Merged 5570dd8c8254a085e10ea88f633e1dee8b00a09f into 44814584f3d296f771122d9cd5fa87336b994f94 )

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
( Success: Merged 63c14cb56ecf0ce015a493071de90a8cb4d60245 into 44814584f3d296f771122d9cd5fa87336b994f94 )

File: lib/cli_host_controller.rb [hunk 1]
```diff
@@ -47,7 +47,20 @@ def lock(host_name)
       raise GLI::BadCommandLine.new("Please provide a host name argument")
     end
 
-    @out.puts "Locking host '#{host_name}' to be implemented..."
-    true
+    host_config = HostConfig.for_directory(@config_dir)
+    host_config.read
+    if !host_config.host(host_name)
+      raise LockError.new("Host name #{host_name} doesn't exist in " +
+        "configuration file")
```
Comment (tgoettlicher) [inline lib/cli_host_controller.rb]: Can you add the location of the config file to the error message?

File: lib/lock_service.rb [hunk 2]
```diff
@@ -0,0 +1,59 @@
+# Copyright (c) 2013-2014 SUSE LLC
```
Comment (tgoettlicher) [inline lib/lock_service.rb]: That's a very minor thing: It's 2015 now. :-)

File: spec/lock_service_spec.rb [hunk 3]
```diff
@@ -0,0 +1,67 @@
+# Copyright (c) 2013-2014 SUSE LLC
```
Comment (tgoettlicher) [inline spec/lock_service_spec.rb]: 2015 again

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
( Success: Merged 36c50316064bf55a85cddbda7b72385e0a62e92c into 44814584f3d296f771122d9cd5fa87336b994f94 )

**Comment by tgoettlicher** [comment]
LGTM

**Comment by cornelius** [comment]
Cleaned up the history and merged.

**Pull request closed without merging**